src/test/java/executor/service/service/ScenarioExecutorImplTest.java

```diff
@@ -0,0 +1,110 @@
+package executor.service.service;
+
+import executor.service.model.configs.WebDriverConfig;
+import executor.service.model.proxy.ProxyConfigHolder;
+import executor.service.model.proxy.ProxyCredentials;
+import executor.service.model.proxy.ProxyNetworkConfig;
+import executor.service.model.scenario.Scenario;
+import executor.service.model.scenario.Step;
+import executor.service.model.scenario.StepTypes;
+import executor.service.service.scenarios.ScenarioExecutor;
+import executor.service.service.scenarios.impl.ScenarioExecutorImpl;
+import executor.service.service.stepexecution.StepExecution;
+import executor.service.service.stepexecution.StepExecutionFabric;
+import executor.service.service.webdriver.WebDriverInitializerImpl;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.openqa.selenium.WebDriver;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.times;
+
+/**
+ * Class for testing the implementation of an ScenarioExecutorImpl.class.
+ *
+ * @author Oleksandr Tuleninov
+ * @version 01
+ */
+public class ScenarioExecutorImplTest {
+
+    private StepExecutionFabric stepExecutionFabric;
+    private StepExecution stepExecution;
+    private Scenario scenario;
+    private WebDriverConfig webDriverConfig;
+
+    @BeforeEach
+    public void setUp() {
+        this.stepExecutionFabric = mock(StepExecutionFabric.class);
+        this.stepExecution = mock(StepExecution.class);
+        this.scenario = getScenario();
+        this.webDriverConfig = getWebDriverConfig();
+    }
+
+    @AfterEach
+    void tearDown() {
+        this.stepExecutionFabric = null;
+        this.scenario = null;
+    }
+
+    @Test
+    public void testInvokeMethodExecute() {
+        ScenarioExecutor scenarioExecutor = mock(ScenarioExecutor.class);
+        doNothing().when(scenarioExecutor).execute(any(Scenario.class), any(WebDriver.class));
+
+        WebDriver webDriver = new WebDriverInitializerImpl(webDriverConfig).getInstance(getProxyConfigHolder());
+        scenarioExecutor.execute(scenario, webDriver);
+
+        verify(scenarioExecutor, times(1)).execute(any(Scenario.class), any(WebDriver.class));
+        verifyNoMoreInteractions(scenarioExecutor);
+    }
+
+    @Test
+    public void testWorkMethodExecute() {
+        when(stepExecutionFabric.getStepExecutor(anyString())).thenReturn(stepExecution);
+        doNothing().when(stepExecution).step(any(WebDriver.class), any(Step.class));
+
+        ScenarioExecutor executionService = new ScenarioExecutorImpl(stepExecutionFabric);
```

Comment:
1)Dont get me confused between ScenarioExecutor and ExecutionService. Please change the name of variable.
2)This is done in init method.